### DISCUSSION TITLE
Porting Rpi pico to esp32

### DISCUSSION THREAD

**Discussion by Jatin1o1** [fhdm-dev/scoppy#121] [General]
I am looking forward to port the source code of rpi pico to esp32 to extend the possiblities of scoppy project.
Since Esp32 is higher performing

**Comment by dhsont** [comment]
esp32 also has wifi and bluetooth, and its dev board is cheaper than rasp pi pico (wifi)

**Comment by niyaz-ahmad** [comment]
Can someone check if this is the source?
   https://github.com/mars-low/scoppy-pico

**Comment by Fabian-Schmidt** [comment]
The issue with the ESP32 is its poor performance of the ADC. Please ensure that the ESP32 matches your expectations.

**Comment by dhsont** [comment]
Any progress to this?  

For ADC we can use external ADC ADS1115 , which is much much more accurate than Arduino. 

we can keep both options with External ADC & without External ADC

**Reply by tyeth** [reply]
I don't think you'll get the accuracy with the esp32's onboard adc (assuming you avoid pin clashes with wifi), or alternatively the sample rate you desire if using the ADS1x15 series, I assume they clock at 1MHz I2C but even if they clock at 4MHz bus speed the overhead will kill your sample speed. Look for a fast SPI based ADC, or use the same one as recommended for pico and save time developing?

That all naively assumes you want what I want which is fast sample rates (>1MHz) and multi-channel at resolutions equivalent to $4000 scopes from the 90s. Not a reasonably assumption on my part, my apologies
